### PR TITLE
ddl: resolve table charset option from collation option

### LINKED ISSUE #13438: Unexpected error when creating table with `Collate` option
## Bug Report

1. What did you do?
If possible, provide a recipe for reproducing the error.
```
mysql> create table t(a int) collate utf8_general_ci;
```
2. What did you expect to see?
Table `t` is created successfully.

3. What did you see instead?
```
ERROR 1115 (42000): Unknown character set: 'utf8mb4'
```

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
This bug exists in all of the latest TiDB release and master versions. 

This bug is related to #10122

### PR DESCRIPTION
**Pull request by bb7133**
### What problem does this PR solve? 
This PR tries to close #13438 

### What is changed and how it works?
When creating/altering table, the charset of table should be resolved from collation. In order to archive this, `getCharsetAndCollateInTableOption` is updated.

Notice that in MySQL, collation is always correlated with charset: when collation is specified, the charset it belongs to is specified as well.

### Check List 

Tests 

 - Integration test

Code changes

 - NA
 
Side effects
 
 - NA

Related changes

 - Need to cherry-pick to the release branch
 
Release note

 - Fix the bug that unexpected error returned when `CREATE TABLE` or `ALTER TABLE` statment has  `COLLATE` option only.

### CODE REVIEW

**Comment by bb7133** [comment]
PTAL @crazycs520 @AilinKid @tangenta @zimulala

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -1815,8 +1790,9 @@ func handleTableOptions(options []*ast.TableOption, tbInfo *model.TableInfo) err
 	if tbInfo.PreSplitRegions > tbInfo.ShardRowIDBits {
 		tbInfo.PreSplitRegions = tbInfo.ShardRowIDBits
 	}
-
-	return nil
+	var err error
+	tbInfo.Charset, tbInfo.Collate, err = getCharsetAndCollateInTableOption(0, options)
+	return err
```
Comment (tangenta) [inline ddl/ddl_api.go]: The enclosing method `handleTableOptions` is called in `buildTableInfoWithCheck`, which has already called `getCharsetAndCollateInTableOption` at line 1312 at this point.  Is the `getCharsetAndCollateInTableOption` here redundant?

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -1830,39 +1806,36 @@ func isIgnorableSpec(tp ast.AlterTableType) bool {
 // and returns the last charset and collate in options. If there is no charset in the options,
 // the returns charset will be "", the same as collate.
 func getCharsetAndCollateInTableOption(startIdx int, options []*ast.TableOption) (ca, co string, err error) {
```
Comment (AilinKid) [inline ddl/ddl_api.go]: ca, co is not obviously clear.

File: ddl/ddl_api.go (hunk 1 above)
Comment (AilinKid) [reply ddl/ddl_api.go]: +1

Comment (bb7133) [reply ddl/ddl_api.go]: You're right, I just forgot to remove those codes.

Thanks!

File: ddl/ddl_api.go (hunk 2 above)
Comment (bb7133) [reply ddl/ddl_api.go]: Sure...`charset` has been used by the `charset` package, so I changed `ca`, `co` to `chs` and `coll` respectively.

**Comment by bb7133** [comment]
Addressed, PTAL @AilinKid @tangenta

**Pull request merged**